build/root/Makefile.generated_files

```diff
@@ -496,3 +497,86 @@ gen_bindata: $(BINDATA_GEN) FORCE
 $(BINDATA_GEN): $(k8s.io/kubernetes/vendor/github.com/jteeuwen/go-bindata/go-bindata)
 	KUBE_BUILD_PLATFORMS="" hack/make-rules/build.sh ./vendor/github.com/jteeuwen/go-bindata/go-bindata
 	touch $@
+
+
+# Register generation
+#
+
+# The result file, in each pkg, of register-gen generation.
+REGISTERGEN_BASENAME := $(GENERATED_FILE_PREFIX)register
+REGISTERGEN_FILENAME := $(REGISTERGEN_BASENAME).go
+REGISTERGEN_OUTPUT_PKG := $(PRJ_SRC_PATH)/vendored
+
+# The tool used to generate register.go files.
+REGISTER_GEN := $(BIN_DIR)/register-gen
+
+# Find all the directories that request register generation.
+# files that need deepcopy also need to generate "register" files
+ifeq ($(DBG_MAKEFILE),1)
+    $(warning ***** finding all +k8s:deepcopy-gen tags)
+endif
+REGISTERGEN_DIRS := $(shell                                                 \
+    find ./staging/src/k8s.io/api -name types.go | xargs -n1 dirname | sort \
```

Comment:
maybe we should be consistent with the other code generators and add a `+k8s:register-gen=true` tag to each of the `types.go` files instead?

Reply:
it seems to be a bit redundant because we can infer the type (external, internal) without the additional tag.

Reply:
I tend to agree with @ixdy.

Reply:
@sttts  doesn't this also change `register-gen` app ? Because we explicitly designed the app to not follow the tags (see: https://github.com/kubernetes/kubernetes/pull/66251#discussion_r202803315)  ?

Reply:
Neither kubebuilder nor operator-sdk use a package tag for register generation. Also, we only pass those packages that need register.go generation. Hence, I think we should go without this additional tag.